### DISCUSSION TITLE
How to only view commits created on a fork or branch?

### DISCUSSION THREAD

**Discussion by iCodeSometime** [community/community#22055] [New to GitHub]
<p>So, if I have a project that was forked, and I want to view only commits added to the fork, and not commits from the original project; how can I do that?</p>
<p>Or more generally, view only commits added on the current branch? e.g. branch ‘feature’ was created from master, and I’d like to see a list of all commits created on feature, but not the commits from master?</p>

**Comment by li-pamela** [comment]
<p>Hi <a class="mention" href="/icodesometime">@icodesometime</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the How to use Git and GitHub board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Cheers!</p>

**Comment by iCodeSometime** [comment]
<p>Thanks… Sorry, this is my first time using the forum</p>

**Comment by lee-dohm** [comment]
<p>You can see the commits that exist only on the target branch but not on the base branch with the following (rather long) command:</p>
<pre><code>git log --decorate --pretty=oneline --abbrev-commit --no-merges --first-parent base..target
</code></pre>
<p>If you want to see what the various options do, you can <a href="https://explainshell.com/explain?cmd=git+log+--decorate+--pretty%3Doneline+--abbrev-commit+--no-merges+--first-parent+base..target#" rel="nofollow noopener">view this command on explainshell.com</a>.</p>
<p>I hope that helps!</p>

**Comment by iCodeSometime** [comment]
<p>Cool, thanks! I gather there’s no way to do this in GitHub then?</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/icodesometime">@icodesometime</a>,</p>
<p>Actually you should be able to view all the commits for a branch on <a href="http://github.com" rel="nofollow noopener">github.com</a>. To do so, you just need to select the branch from the “Branch” dropdown.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/47c8dc920cde7fb52051424b94deb2929dac0666.png" data-download-href="/uploads/short-url/af2f9u1mij6Nf8PBWsMJxkG1qd0.png?dl=1" title="branch dropdown.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16922726/181068725-4b62c247-cc20-4f00-8276-4934b97d0892.png" alt="branch dropdown.png" data-base62-sha1="af2f9u1mij6Nf8PBWsMJxkG1qd0" width="689" height="74" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/47c8dc920cde7fb52051424b94deb2929dac0666_2_689x74.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/47c8dc920cde7fb52051424b94deb2929dac0666_2_1033x111.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/47c8dc920cde7fb52051424b94deb2929dac0666.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/47c8dc920cde7fb52051424b94deb2929dac0666_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branch dropdown.png</span><span class="informations">1292×139 17.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This will put you directly on the branch whose commits you want to view. Then you can select “Compare”</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d98c36422f6dc14fac37ee82c3ac33e7656f0d53.png" data-download-href="/uploads/short-url/v2w0R2LpBWdw8I4ot9pUIgYUzUT.png?dl=1" title="Compare.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16922726/181068727-a94ae7e7-123f-4acd-b7a0-375971fefa8b.png" alt="Compare.png" data-base62-sha1="v2w0R2LpBWdw8I4ot9pUIgYUzUT" width="689" height="93" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d98c36422f6dc14fac37ee82c3ac33e7656f0d53_2_689x93.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d98c36422f6dc14fac37ee82c3ac33e7656f0d53_2_1033x139.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d98c36422f6dc14fac37ee82c3ac33e7656f0d53.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d98c36422f6dc14fac37ee82c3ac33e7656f0d53_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Compare.png</span><span class="informations">1308×178 25.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This will allow you to view a list of all the commits made for that branch.</p>
<p>Hope this helps! Cheers!</p>

**Reply by simon-grantham** [reply]
This didn't really answer the question. I think the questioner just gave up.  The question included the word "latest".  I believe the person was asking about the sort order.  When github shows the commit history for a branch, it does so from oldest to latest.  What is the point of that?  How many people want to repeat "page down" that many times vs how many people actually want to see commits from years ago?

**Comment by JacekPliszka** [comment]
<p>What if I want to view last commits to master?</p>
<p>Compare does not show on master…</p>

**Comment by VictorZZZZ** [comment]
<p>On github you can go to repository you need, and then append to the address string the<br>
<code>.../commits/branch-name</code> line</p>
<p>So you will get something like this:</p>
<p><code> https://github.com/some-account/some-repository/commits/branch-name</code></p>

**Reply by niloct** [reply]
And then you can also filter by a commiter's `username` on the right-side select box.

**Comment by JosepeDevs** [comment]
I was looking for my complete historic, of a specific branch, to access, enter the repo, select the branch and at the right  click "Activity", there you can navigate all commits to that branch

**Reply by TonyAbouHaidar** [reply]
any idea if this is supported in Github Desktop?  the history tab seems to show commits from all branches ..
